src/module.rs

```diff
@@ -30,41 +30,96 @@ impl WasmModule {
         }
     }
 
+    pub fn imports(&self) -> impl Iterator<Item = &ImportEntry> {
+        self.module
+            .import_section()
+            .map_or(Either::Left(iter::empty()),
+                    |section| Either::Right(section.entries().iter()))
+    }
+
+    /// Safe function for counting the number of imported functions.
+    /// Use this instead of `self.imports().size_hint()`!
+    pub fn count_imported_functions(&self) -> usize {
+        self.module.import_count(ImportCountType::Function)
+    }
+
+    pub fn imported_functions(&self) -> impl Iterator<Item = WasmFunction> {
+        self.imports()
+            .enumerate()
+            .filter_map(move |(i, import)| if let External::Function(tyid) = import.external() {
```

Comment:
Okay I looked into this more deeply.

By incrementing `i` for each element in `self.imports()` we end up incrementing for both functions and globals (since they both exist in `self.imports()`). I'm pretty sure this clause only executes if `import` is a function rather than a global (based off of my tests with this code). This means that if we have 12 functions and 2 globals, we can end up with an iterator that has 12 elements, but the last element has an id of 13 because 2 globals were omitted.

This is confusing to me. Do we have a good reason that this function only returns an iterator of `WasmFunction` rather than all kinds of imports? Alternatively, if we want to only return `WasmFunction` maybe we could return a map of indexes as to not hide jumps in ids in the iterator.

Reply:
Good catch, thanks! Fixed in the update.